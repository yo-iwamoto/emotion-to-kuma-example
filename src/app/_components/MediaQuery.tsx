import { css } from '@kuma-ui/core';

export function MediaQuery() {
  return <p className={styles.text}>MediaQuery</p>;
}

const styles = {
  text: css`
    color: orange;
    font-size: 32px;
    @media screen and (sm <= width < md) {
      font-size: 36px;
    }
    @media screen and (md <= width) {
      font-size: 40px;
    }
  `,
};
