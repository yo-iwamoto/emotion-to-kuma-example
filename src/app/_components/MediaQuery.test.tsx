import { MediaQuery } from './MediaQuery';
import { render } from '@testing-library/react';

describe('MediaQuery', () => {
  it('renders unchanged', () => {
    const { asFragment } = render(<MediaQuery />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('has color style', () => {
    const { getByText } = render(<MediaQuery />);

    expect(getByText('MediaQuery')).toHaveStyleRule('color', 'orange');
  });
});
