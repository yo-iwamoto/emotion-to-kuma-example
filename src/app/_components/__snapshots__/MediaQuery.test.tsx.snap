// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MediaQuery > renders unchanged 1`] = `
<DocumentFragment>
  .🐻-155337860 {
  color: orange;
  font-size: 32px;
}

@media screen and (375px &lt;= width &lt; 800px) {
  .🐻-155337860 {
    font-size: 36px;
  }
}

@media screen and (800px &lt;= width) {
  .🐻-155337860 {
    font-size: 40px;
  }
}

  <p
    class="🐻-155337860"
  >
    MediaQuery
  </p>
</DocumentFragment>
`;
